contrib/cassandra-connector/src/main/java/com/google/cloud/cassandra/dataflow/io/CassandraIO.java

```diff
@@ -0,0 +1,307 @@
+import com.google.cloud.dataflow.sdk.coders.SerializableCoder;
+import com.google.cloud.dataflow.sdk.transforms.Create;
+import com.google.cloud.dataflow.sdk.transforms.DoFn;
+import com.google.cloud.dataflow.sdk.transforms.PTransform;
+import com.google.cloud.dataflow.sdk.transforms.ParDo;
+import com.google.cloud.dataflow.sdk.transforms.View;
+import com.google.cloud.dataflow.sdk.values.PCollection;
+import com.google.cloud.dataflow.sdk.values.PCollectionView;
+import com.google.cloud.dataflow.sdk.values.PDone;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.common.util.concurrent.Uninterruptibles;
+
+public class CassandraIO {
+
+	/**
+	 * Write sink for Cassandra.
+	 */
+	public static class Write {
+
+		/** create CassandraIO.Write.Unbound with hosts */
+		public static Unbound hosts(String... hosts) {
+			return new Unbound().hosts(hosts);
+		}
+
+		/** create CassandraIO.Write.Unbound with keyspace */
+		public static Unbound keyspace(String keyspace) {
+			return new Unbound().keyspace(keyspace);
+		}
+
+		/** create CassandraIO.Write.Unbound with port */
+		public static Unbound port(int port) {
+			return new Unbound().port(port);
+		}
+
+		/** Set cassandra hosts,keyspace and port */
+		public static class Unbound {
+
+			private final String[] _hosts;
+			private final String _keyspace;
+			private final int _port;
+
+			Unbound() {
+				_hosts = null;
+				_keyspace = null;
+				_port = 9042;
+			}
+
+			Unbound(String[] hosts, String keyspace, int port) {
+				_hosts = hosts;
+				_keyspace = keyspace;
+				_port = port;
+			}
+
+			/** create CassandraIO.Write.Bound */
+			public <T> Bound<T> bind() {
+				return new Bound<T>(_hosts, _keyspace, _port);
+			}
+
+			/** create CassandraIO.Write.Unbound with hosts */
+
+			public Unbound hosts(String... hosts) {
+				return new Unbound(hosts, _keyspace, _port);
+			}
+
+			/** create CassandraIO.Write.Unbound with keyspace */
+			public Unbound keyspace(String keyspace) {
+				return new Unbound(_hosts, keyspace, _port);
+			}
+
+			/** create CassandraIO.Write.Unbound with port */
+			public Unbound port(int port) {
+				return new Unbound(_hosts, _keyspace, port);
+			}
+		}
+
+		/**
+		 * Set Cassandra hosts,keyspace and port Apply transformation
+		 */
+		public static class Bound<T> extends PTransform<PCollection<T>, PDone> {
+
+			private static final long serialVersionUID = 0;
+
+			private final String[] _hosts;
+			private final String _keyspace;
+			private final int _port;
+
+			Bound(String[] hosts, String keyspace, int port) {
+				_hosts = hosts;
+				_keyspace = keyspace;
+				_port = port;
+			}
+
+			public String[] getHosts() {
+				return _hosts;
+			}
+
+			public String getKeyspace() {
+				return _keyspace;
+			}
+
+			public int getPort() {
+				return _port;
+			}
+
+			/**
+			 * Create CassandraWriteOperation Create Coder write entity
+			 */
+			@SuppressWarnings("unchecked")
+			public PDone apply(PCollection<T> input) {
```

Comment:
This method of writing a sink has several major problems, and would be better done with the [Sink API](https://github.com/apache/incubator-beam/blob/master/sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/io/Sink.java)

The main things I see to avoid are
- The CassandraWriteOperations are not shared between your writing DoFn and the finalizing DoFn when this is run on a distributed backend, so none of the session closing logic will work properly
- You are flushing the writes after every element which is likely to be inefficient
- There is no way for the user to control the format in which the records are written